jena-arq/src/main/java/org/apache/jena/riot/RDFParserBuilder.java

```diff
@@ -504,8 +504,10 @@ private HttpClient buildHttpClient() {
             Header header = new BasicHeader(k, v);
             hdrs.add(header);
         });
-        HttpClient hc = CachingHttpClientBuilder.create().setDefaultHeaders(hdrs).build();
-        return hc;
+        HttpClient hc = HttpOp.createPoolingHttpClientBuilder()
```

Comment:
What is required is a builder that is based on another but isolated from it by cloning (or in some way) so that adding new builder settings only changes the clone, not also the original (the current active default in `HttpOp` in this case).`HttpClientBuilder` is a setter only style, you can't get settings from it (builder builder needed ?!). `RDFparserBuilder` has `clone()`.